client/sttp-client/src/main/scala/sttp/tapir/client/sttp/EndpointToSttpClient.scala

```diff
@@ -62,15 +63,45 @@ private[sttp] class EndpointToSttpClient[R](clientOptions: SttpClientOptions, ws
           case EndpointIO.FixedHeader(_, codec, _)         => codec.decode(())
           case EndpointIO.Empty(codec, _)                  => codec.decode(())
           case EndpointOutput.OneOf(mappings, codec) =>
-            mappings
-              .find(mapping => mapping.statusCode.isEmpty || mapping.statusCode.contains(meta.code)) match {
-              case Some(mapping) =>
-                getOutputParams(mapping.output, body, meta).flatMap(p => codec.decode(p.asAny))
+            val mappingsForStatus = mappings collect {
+              case m if m.statusCode.isEmpty || m.statusCode.contains(meta.code) => m
+            }
+
+            val contentType = meta
+              .header(HeaderNames.ContentType)
+              .map(MediaType.parse)
+
+            def applyMapping(m: EndpointOutput.StatusMapping[_]) = getOutputParams(m.output, body, meta).flatMap(p => codec.decode(p.asAny))
+
+            contentType match {
               case None =>
-                DecodeResult.Error(
-                  meta.statusText,
-                  new IllegalArgumentException(s"Cannot find mapping for status code ${meta.code} in outputs $output")
-                )
+                mappingsForStatus.headOption
```

Comment:
shouldn't we give priority to mappings without a body in such case? e.g. for a mapping
```
oneOf[Either[Unit, X]](
  (Ok, jsonBody[X].map(Right(_)),
  (Ok, emptyOutput.map(Left(_))
)
```

in case we get back an empty response, the second should be picked